enumerables.rb

```diff
@@ -0,0 +1,102 @@
+module Enumerable
+  def my_each
+    size = count - 1
+    counter = 0
+    items = to_a
+    while counter <= size
+      if is_a? Range
+        items.length >= 0 && yield(items[counter])
+      elsif is_a? Hash
+        yield(items[counter][0], items[counter][1])
+      else
+        yield(self[counter])
+      end
+      counter += 1
+    end
+    self
+  end
+
+  def my_each_with_index()
+    counter = 0
+    size = length - 1
+    while counter <= size
+      yield(self[counter], counter)
+      counter += 1
+    end
+    self
+  end
+
+  def my_select
+    required = []
+    my_each do |r|
+      required.push(yield(r))
+    end
+    required
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns an enumerator if no block is given
```ruby
array.my_select == Enumerator #true
```
- [ ] returns an array containing all elements of enum for which the given block returns a true value
```ruby
array = Array.new(100){rand(0...9)}
array.my_select(&block) == array.select(&block) #true
```
```

Reply:
Hi @whiz25, please endeavor to run these cases in your terminal before your next feedback,
just take a look at this screenshot you'd see that 
`array.my_select == Enumerator #true` actually is `false`

![image](https://user-images.githubusercontent.com/46686100/75714189-87855480-5ccb-11ea-9a7d-76647ee33e59.png)